core/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -1144,6 +1098,10 @@ public static MetaData fromXContent(XContentParser parser) throws IOException {
                         while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
                             builder.put(IndexTemplateMetaData.Builder.fromXContent(parser, parser.currentName()));
                         }
+                    } else if ("warmers".equals(currentFieldName)) {
+                        // TODO: remove this condition in 4.0
```

Comment:
Add an assertion to this file that major is less than 4.0 with a comment about this todo?